src/auth/session/storage/sqlite.ts

```diff
@@ -0,0 +1,114 @@
+import sqlite3 from 'sqlite3';
+
+import {SessionInterface} from '../types';
+import {SessionStorage} from '../session_storage';
+import {sessionFromEntries, sessionEntries} from '../session';
+
+export interface SQLiteSessionStorageOptions {
+  createDBWhenMissing: boolean;
+  sessionTableName: string;
+}
+const defaultMySQLSessionStorageOptions: SQLiteSessionStorageOptions = {
+  createDBWhenMissing: true,
+  sessionTableName: 'shopify_node_api_sessions',
+};
+
+export class SQLiteSessionStorage implements SessionStorage {
+  private options: SQLiteSessionStorageOptions;
+  private db: sqlite3.Database;
+  private ready: Promise<void>;
+
+  constructor(
+    private filename: string,
+    opts: Partial<SQLiteSessionStorageOptions> = {},
+  ) {
+    this.options = {...defaultMySQLSessionStorageOptions, ...opts};
+    this.db = new sqlite3.Database(this.filename);
+    this.ready = this.init();
+  }
+
+  public async storeSession(session: SessionInterface): Promise<boolean> {
```

Comment:
nit: is `saveSession` a better public method?

Reply:
No strong opinion from my end. Let’s have this discussion, but address it in a separate PR. The `SessionStorage` interface predates me :D

Reply:
Yeah, `loadSession` and `storeSession` are from the original release of the Node API library ... don't know why we named them as such (instead of `getSession` and `saveSession`, for example!) but this would be a breaking change (therefore major version change), whereas adding the `SessionStorage` adapters for different databases _might_ be a feature addition (minor change), although arguably significant enough to be major too?

Agree with @surma, separate discussion and PR required.

Reply:
Huh ... just noticed that `loadSession`, `storeSession` and `deleteSession` were part of PR#1 for this repo, i.e., these interface method were literally the first code added!

Reply:
yeah on second thought, I think the name is fine